ansible/controller_files/playbook.yml

```diff
@@ -0,0 +1,68 @@
+- name: Deploy ToDo app
+  hosts: webservers
+  remote_user: ec2-user
+
+  vars_prompt:
+    - name: secret_key
+      private: true
+      prompt: Enter the Flask secret key
+      salt_size: 8
+
+    - name: trello_key
+      private: true
+      prompt: Enter the Trello key
+      salt_size: 8
+    
+    - name: trello_token
+      private: true
+      prompt: Enter the Trello token
+      salt_size: 8   
+
+  tasks:
+    - name: Install yum Packages
+      yum:
+        name:
+          - git
+          - python3
+      become: true
+
+    - name: Install poetry
+      ansible.builtin.shell: curl -sSL https://install.python-poetry.org | python3 -
+      args:
+        creates: ~/.local/bin/poetry
+      become: true
+
+    - name: Create To-Do app directory
+      ansible.builtin.file:
+        path: /opt/todoapp
+        state: directory
+        mode: "777"
```

Comment:
Agree this will work but is quite permissive, it would be nice to try restricting this a bit more, for example setting the ec2-user as the owner and restricting write/execute permissions to other users;
```
      owner: ec2-user
      mode: '744'
```
I haven't tested this against your future commands so I don't guarantee it works exactly as is though!